### ISSUE TITLE
netcdf on more than one core throws errors...

### ISSUE THREAD

**Issue by xy124** [parflow/parflow#97]
Netcdf's when using more than one core seem wrong!
```bash
compare_nc.py default_richards.out.00000.nc tmp/default_richards.out.00000.nc 
they differ in: pressure
difference:
3.9876839873547476e+39
Max difference:
9.969209968386869e+36
they differ in: saturation
difference:
3.9876839873547476e+39
Max difference:
9.969209968386869e+36
```
both nc files are created using standard testcases with the latest parflow master version  (1bae2ce40df746917539535660ad22b7d69e1b2c - but the bug appeared also in older versions):

```bash
docker@4b4ee798a74c:~/parflow/test$ tclsh default_richards_with_netcdf.tcl
```
when running on more than one core we get this on the console out:
```bash
[4b4ee798a74c:08824] Read -1, expected 5160, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1
[4b4ee798a74c:08824] Read -1, expected 5120, errno = 1


default_richards_with_netcdf : PASSED
```

Probably this is a duplicate of #49 .
But it should be easy to reproduce:
just do the steps provided in the https://github.com/xy124/containered-parflowvr/ Readme and install the latest parflow in it:
```bash
docker@4b4ee798a74c:~/$ rm -rf parflow
docker@4b4ee798a74c:~/$ git clone https://github.com/parflow/parflow
cd parflow
mkdir build
cd build
ccmake ..  ##and activate netcdf, hypre, amps->mpi1, set CMAKE_INSTALL_PREFIX to /home/docker/install
make install
cd ../test
## ... run default_richards_with_netcdf.tcl testcases with different node configurations
```

**Comment by xy124** [comment]
I found out that the strange log messages (`[4b4ee798a74c:08824] Read -1, expected 5160, errno = 1`...) are relied to openMPI which is run a docker container and thus can be avoided as stated on https://github.com/open-mpi/ompi/issues/4948 by setting e.g. `export OMPI_MCA_btl="^vader"`

Still the netcdf files are broken!

**Comment by kulkarni1** [comment]
I think its still not a netCDF issue. Are you running parflow inside docker
image on single host?

**Comment by xy124** [comment]
At the moment yes. But everything else on multiple cores works just fine. Also in the Docker image.
To me it looks like duplicate of #49 (but here we can see exactly which steps provoce the error...). What is a proper way to use netCDF with the cmake build system of Parflow? Are you using the netCDF version that is shipped with your distribution? Or are you always compiling by hand? Are you isntalling with pnetcdf or hdf5?

**Comment by kulkarni1** [comment]
Yeah this is a bit puzzling. For CMake, its similar to other options and
nothing special. While testing, I had tried system installed netCDF libs
and self compiled libs also and never faced this issue. Is it possible for
you to test it outside docker environment on your machine?

**Comment by xy124** [comment]
As I need it to work in the docker image that would not make so much sense...

**Comment by kulkarni1** [comment]
can you compile netCDF and HDF5 with mpi support in docker and try

**Comment by xy124** [comment]
I solved the problem now when compiling netCDF with configure and yes, HDF5.
It is still really strange that parflow is depending that hard on which compile options were used for netCDF (and especially that one **can** compile Parflow with netcdf if the **WRONG** compile options were used for building netcdf.) (Actually I can also imagine that in parflow sth is done wrong. The wrong netcdf's looked a bit as if only one core wrote its subvectors to the netcdf and the others left everything in the netcdf to the default value - if somebody wants to debug this there is the docker container up there to reproduce ;) use commit 0458e5e4b39a4a540fc4c42a35b2c01813a29281 as the later ones are fixing it using netcdf compiled with configure)

**Comment by kulkarni1** [comment]
what compile options are you using? while compiling HDF5 and netCDF to
enable paralle I/O functionality, I had not used any exotic options. Just
plain vanilla installation. I am still bit confused about error you are
facing.
